src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -264,28 +264,38 @@ const connectInfiniteHits: InfiniteHitsConnector = (
       getWidgetState(uiState, { searchParameters }) {
         const page = searchParameters.page || 0;
 
-        if (!hasShowPrevious || page === 0 || page + 1 === uiState.page) {
+        if (!hasShowPrevious || !page) {
           return uiState;
         }
 
         return {
           ...uiState,
+          // The page in the UI state is incremented by one
+          // to expose the user value (not `0`).
           page: page + 1,
         };
       },
 
       getWidgetSearchParameters(searchParameters, { uiState }) {
-        if (!hasShowPrevious) {
-          return searchParameters;
+        let widgetSearchParameters = searchParameters;
+
+        if (escapeHTML) {
+          widgetSearchParameters = searchParameters.setQueryParameters(
+            TAG_PLACEHOLDER
+          );
         }
 
-        const uiPage = uiState.page;
+        if (hasShowPrevious) {
+          const uiPage = uiState.page;
 
-        if (uiPage) {
-          return searchParameters.setQueryParameter('page', uiPage - 1);
+          if (uiPage) {
+            // The page in the search parameters is decremented by one
+            // to get to the actual parameter value from the UI state.
+            return widgetSearchParameters.setQueryParameter('page', uiPage - 1);
```

Comment:
We don't check for negative values etc.. but I guess it's fine for now. We never have any issues about that.